### PR TITLE
Support python Transformer and add Transformer example

### PR DESCRIPTION
**Pull request by dding3**
1. Add document for `Transformer`
2. Add python `Transformer`
3. Add `Transformer` example

This PR is dependent on https://github.com/intel-analytics/analytics-zoo/pull/966 and https://github.com/intel-analytics/analytics-zoo/pull/1218

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/keras/layers/transformer.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from zoo.pipeline.api.keras.layers import *
+from zoo.pipeline.api.keras.models import Sequential
+
+
+if sys.version >= '3':
+    long = int
+    unicode = str
+
+
+class TransformerLayer(ZooKerasLayer):
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/keras/layers/transformer.py]: don't wrap everything in Java; instead, build the layer in python

Comment (dding3) [reply pyzoo/zoo/pipeline/api/keras/layers/transformer.py]: Updated

File: pyzoo/zoo/pipeline/api/autograd.py [hunk 2]
```diff
@@ -484,6 +484,21 @@ def set_weight(self, value):
                       kbase.JTensor.from_ndarray(value))
 
 
+class Constant(kbase.ZooKerasLayer, VariableOperator):
+    """
+    A un-trainable Variable without weights.
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/autograd.py]: A constant Variable

**Comment by dding3** [comment]
jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2031/

**Comment by jason-dai** [comment]
LGTM

**Comment by dding3** [comment]
Thanks for your review. 
Jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2040/

**Pull request merged**